packages/hardware-ledger/src/transformers/assets.ts

```diff
@@ -0,0 +1,44 @@
+import * as Ledger from '@cardano-foundation/ledgerjs-hw-app-cardano';
+import { Cardano } from '@cardano-sdk/core';
+
+const sortTokensCanonically = (tokens: Ledger.Token[]) => {
+  tokens.sort((a, b) => {
+    if (a.assetNameHex === b.assetNameHex) {
+      return a.assetNameHex > b.assetNameHex ? 1 : -1;
+    } else if (a.assetNameHex.length > b.assetNameHex.length) return 1;
+    return -1;
+  });
+};
+
+const tokenMapToAssetGroup = (tokenMap: Cardano.TokenMap): Ledger.AssetGroup[] => {
+  const map = new Map<string, Array<Ledger.Token>>();
+
+  for (const [key, value] of tokenMap.entries()) {
+    const policyId = Cardano.AssetId.getPolicyId(key);
+    const assetName = Cardano.AssetId.getAssetName(key);
+
+    if (!map.has(policyId)) map.set(policyId, new Array<Ledger.Token>());
+
+    map.get(policyId)!.push({
+      amount: value,
+      assetNameHex: assetName
+    });
+  }
+
+  const tokenMapAssetsGroup = [];
+  for (const [key, value] of map.entries()) {
+    sortTokensCanonically(value);
```

Comment:
This is fine as it's an internal implementation detail, just want to note that we prefer this style (typically don't do procedural)
```suggestion
    value.sort(compareCanonically);
```